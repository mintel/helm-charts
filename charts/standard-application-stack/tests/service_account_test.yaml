suite: Test ServiceAccount Creation
templates:
- service-accounts.yaml
release:
  namespace: test-namespace
tests:
- it: ensures annotations are populated
  set:
    serviceAccount:
      create: true
      annotations:
        test: value
      argo:
        hook: PreSync
        hookDeletePolicy: HookSucceeded
        syncWave: "-1"
  asserts:
  - matchSnapshot: {}
  - hasDocuments:
      count: 1
  - equal:
      path: metadata.annotations["argocd.argoproj.io/hook"]
      value: PreSync
  - equal:
      path: metadata.annotations["argocd.argoproj.io/hook-delete-policy"]
      value: HookSucceeded
  - equal:
      path: metadata.annotations["argocd.argoproj.io/sync-wave"]
      value: "-1"
  - equal:
      path: metadata.annotations.test
      value: value
