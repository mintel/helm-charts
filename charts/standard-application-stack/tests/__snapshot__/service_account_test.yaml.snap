ensures annotations are populated:
  1: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/hook: PreSync
        argocd.argoproj.io/hook-delete-policy: HookSucceeded
        argocd.argoproj.io/sync-wave: "-1"
        test: value
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: example-app
      name: example-app
      namespace: test-namespace
